external-crates/move/move-compiler/src/typing/translate.rs

```diff
@@ -2265,3 +2267,161 @@ fn make_arg_types<S: std::fmt::Display, F: Fn() -> S>(
     }
     given
 }
+
+//**************************************************************************************************
+// Module-wide warnings
+//**************************************************************************************************
+
+/// Generates warnings for unused struct types and unused (private) functions.
+fn gen_unused_warnings(context: &mut Context, ident: ModuleIdent, mdef: &T::ModuleDefinition) {
+    let mut packed_types = BTreeSet::new();
+    let mut called_fns = BTreeSet::new();
+
+    for (_, _, fun) in &mdef.functions {
+        let sp!(_, body) = &fun.body;
+        if let T::FunctionBody_::Defined(seq) = body {
+            for item in seq {
+                let sp!(_, seq_item) = item;
+                seq_item_collect_used(&seq_item, &ident, &mut called_fns, &mut packed_types)
```

Comment:
Why not just collect this during typing itself? Like I don't much of a point as making it a separate pass?

Reply:
Sure. I wanted to be as non-invasive as I could but perhaps too much. Will redo!

Reply:
So, I think you will have to do something like below as a separate pass, just given how the warning suppression works. But I would be fine with collecting the used function calls during typing.

Reply:
An unfortunate consequence of collecting this data during typing is that we won't have this info collected for library (non-source) modules... I added suppression of these new warnings for non-source modules, but I open to other suggestions on how to handle this.